cmd/nerdctl/build_test.go

```diff
@@ -44,6 +44,7 @@ CMD ["echo", "nerdctl-build-test-string"]
 
 	base.Cmd("build", "-t", imageName, buildCtx).AssertOK()
 	base.Cmd("build", buildCtx, "-t", imageName).AssertOK()
+	base.Cmd("build", buildCtx, "-t", imageName, fmt.Sprintf("--output=type=docker,name=name")).AssertOK()
```

Comment:
We should test `base.Cmd("run", "--rm", imageName).AssertOutExactly("nerdctl-build-test-string\n")` again (but the imageName should be changed)

Reply:
Thanks for reviewing, Should we not take what the docker behavior is?

```
➜  nerdctl git:(fix-nerdctl-build) ✗ docker build -t nerdctl --output=type=docker,name="nerdctl2"  .                
➜  nerdctl git:(fix-nerdctl-build) ✗ docker images
REPOSITORY   TAG       IMAGE ID   CREATED   SIZE
....
➜  nerdctl git:(fix-nerdctl-build) ✗ docker images | grep nerdctl
nerdctl      latest    514a49968fa2   3 minutes ago   686MB

➜   git:(main) ✗ docker run -it --rm nerdctl2 /bin/sh
Unable to find image 'nerdctl2:latest' locally
```

Reply:
Can you review again; I added the docker compatibility test to ensure that this fails as well? If you still prefer having a second tag; I will make the code changes appropriately.